src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+            this->actionFunc = func_8088B5F4;
+        } else {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 15.0f;
+            this->actionFunc = func_8088B90C;
+            this->dyna.actor.flags &= ~0x30;
+        }
+
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        Audio_PlayActorSound2(
+            &this->dyna.actor,
+            (SurfaceType_GetSfx(&globalCtx->colCtx, this->dyna.actor.floorPoly, this->dyna.actor.floorBgId) + 0x800) &
+                0xFFFF);
+    }
+
+    this->unk_16C -= 0.5f;
+    this->unk_16C = (this->unk_16C < 0.0f) ? (0.0f) : this->unk_16C;
+
+    if (this->params == 0) {
+        if (func_8004356C(&this->dyna) != 0) {
+            if (this->unk_169 == 0) {
+                this->unk_169 = 3;
+            }
+            Camera_ChangeSetting(globalCtx->cameraPtrs[0], 0x30);
+        } else if (func_8004356C(&this->dyna) == 0) {
+            if (this->unk_169 != 0) {
+                Camera_ChangeSetting(globalCtx->cameraPtrs[0], 3);
+            }
+            this->unk_169 = 0;
+        }
+    }
+}
+
+void func_8088B90C(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 1.0f) != 0) {
+        this->actionFunc = func_8088B634;
+    }
+}
+
+void func_8088B954(BgHidanRock* this, GlobalContext* globalCtx) {
+    s16 unk_16A;
+
+    unk_16A = this->unk_16A;
+    if (unk_16A != 0) {
+        this->unk_16A = unk_16A - 1;
+        unk_16A = this->unk_16A;
+    }
+
+    if (unk_16A == 0) {
+        this->actionFunc = func_8088B79C;
+        this->dyna.actor.velocity.y = 0.0f;
+    }
+}
+
+void func_8088B990(BgHidanRock* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->unk_16A++;
+    if (this->dyna.unk_150 != 0.0f) {
+        this->dyna.actor.speedXZ = 0.0f;
+        player->stateFlags2 &= ~0x10;
+    }
+
+    if ((this->params == 0 && (Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 1820.0f,
+                                                  0.25f, 20.0f, 0.5f) < 0.1f)) ||
+        ((this->params != 0) && (Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 480.0,
+                                                    0.25f, 20.0f, 0.5f) < 0.1f))) {
+        if (this->params == 0) {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        }
+        this->unk_16A = 0x14;
+        this->actionFunc = func_8088B954;
+    }
+
+    this->unk_16C = (((this->dyna.actor.world.pos.y + 50.0f) - this->dyna.actor.home.pos.y) + 40.0f) / 80.0f;
+    if (this->params == 0) {
+        if (func_8004356C(&this->dyna) != 0) {
+            if (this->unk_169 == 0) {
+                this->unk_169 = 3;
+            }
+            Camera_ChangeSetting(globalCtx->cameraPtrs[0], 0x30);
+            return;
+        }
+        if (func_8004356C(&this->dyna) == 0) {
```

Comment:
```suggestion
        } else if (!func_8004356C(&this->dyna)) {
```
Make sure the braces match here.